pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -244,7 +244,8 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 			logrus.Infof("Create done, Updating cluster [%s]", cluster.Name)
 			apiEndpoint, serviceAccountToken, caCert, err = p.driverUpdate(cluster, *spec)
 		}
-	} else if spec.RancherKubernetesEngineConfig.RotateCertificates != nil {
+	} else if spec.RancherKubernetesEngineConfig != nil &&
+		spec.RancherKubernetesEngineConfig.RotateCertificates != nil {
```

Comment:
why this change? how is it related to gke options?
if you are fixing a general nil pointer bug, thats fine, but it shoiuld be a separate commit

Reply:
Yeah, I am fixing a nil pointer bug, I will add a commit separately, thanks!

Reply:
This bug has been fixed in the master, so I drop it.